core/trino-main/src/test/java/io/trino/operator/scalar/TestMathFunctions.java

```diff
@@ -47,1547 +50,3511 @@
+                .isEqualTo((byte) -123);
+
+        assertThat(assertions.function("ceil", "CAST(NULL AS TINYINT)"))
+                .isNull(TINYINT);
+
+        assertThat(assertions.function("ceil", "SMALLINT '123'"))
+                .isEqualTo((short) 123);
+
+        assertThat(assertions.function("ceil", "SMALLINT '-123'"))
+                .isEqualTo((short) -123);
+
+        assertThat(assertions.function("ceil", "CAST(NULL AS SMALLINT)"))
+                .isNull(SMALLINT);
+
+        assertThat(assertions.function("ceil", "123"))
+                .isEqualTo(123);
+
+        assertThat(assertions.function("ceil", "-123"))
+                .isEqualTo(-123);
+
+        assertThat(assertions.function("ceil", "CAST(NULL AS INTEGER)"))
+                .isNull(INTEGER);
+
+        assertThat(assertions.function("ceil", "BIGINT '123'"))
+                .isEqualTo(123L);
+
+        assertThat(assertions.function("ceil", "BIGINT '-123'"))
+                .isEqualTo(-123L);
+
+        assertThat(assertions.function("ceil", "12300000000"))
+                .isEqualTo(12300000000L);
+
+        assertThat(assertions.function("ceil", "-12300000000"))
+                .isEqualTo(-12300000000L);
+
+        assertThat(assertions.function("ceil", "CAST(NULL as BIGINT)"))
+                .isNull(BIGINT);
+
+        assertThat(assertions.function("ceil", "123.0E0"))
+                .isEqualTo(123.0);
+
+        assertThat(assertions.function("ceil", "-123.0E0"))
+                .isEqualTo(-123.0);
+
+        assertThat(assertions.function("ceil", "123.45E0"))
+                .isEqualTo(124.0);
+
+        assertThat(assertions.function("ceil", "-123.45E0"))
+                .isEqualTo(-123.0);
+
+        assertThat(assertions.function("ceil", "CAST(NULL as DOUBLE)"))
+                .isNull(DOUBLE);
+
+        assertThat(assertions.function("ceil", "REAL '123.0'"))
+                .isEqualTo(123.0f);
+
+        assertThat(assertions.function("ceil", "REAL '-123.0'"))
+                .isEqualTo(-123.0f);
+
+        assertThat(assertions.function("ceil", "REAL '123.45'"))
+                .isEqualTo(124.0f);
+
+        assertThat(assertions.function("ceil", "REAL '-123.45'"))
+                .isEqualTo(-123.0f);
+
+        assertThat(assertions.function("ceiling", "12300000000"))
+                .isEqualTo(12300000000L);
+
+        assertThat(assertions.function("ceiling", "-12300000000"))
+                .isEqualTo(-12300000000L);
+
+        assertThat(assertions.function("ceiling", "CAST(NULL AS BIGINT)"))
+                .isNull(BIGINT);
+
+        assertThat(assertions.function("ceiling", "123.0E0"))
+                .isEqualTo(123.0);
+
+        assertThat(assertions.function("ceiling", "-123.0E0"))
+                .isEqualTo(-123.0);
+
+        assertThat(assertions.function("ceiling", "123.45E0"))
+                .isEqualTo(124.0);
+
+        assertThat(assertions.function("ceiling", "-123.45E0"))
+                .isEqualTo(-123.0);
```

Comment:
IntelliJ IDEA isn't happy about this and indicates a warning

```
'isEqualTo()' between objects of inconvertible types 'double' and 'ExpressionAssertProvider' 
```

this is obviously its fault, not the code's, but i still don't want people to start ignoring IDE warnings. They are helpful.

do you have any advice here?